src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c

```diff
@@ -21,25 +33,181 @@ const ActorInit Bg_Jya_Goroiwa_InitVars = {
     (ActorFunc)BgJyaGoroiwa_Update,
     (ActorFunc)BgJyaGoroiwa_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Goroiwa/func_80897970.s")
+extern Gfx D_060006B0[];
+
+static ColliderJntSphItemInit sJntSphItemsInit[] = {
+    {
+        { 0x00, { 0x20000000, 0x00, 0x04 }, { 0x00000000, 0x00, 0x00 }, 0x01, 0x00, 0x01 },
+        { 0, { { 0, 0, 0 }, 58 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x11, 0x00, 0x39, 0x20, COLSHAPE_JNTSPH },
+    1,
+    sJntSphItemsInit,
+};
+
+static CollisionCheckInfoInit collisionCheckInfoInit = { 0x01, 0xF, 0x0, 0xFE };
+
+static InitChainEntry initChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+void func_80897970(BgJyaGoroiwa* this) {
+    Sphere16* worldSphere = &this->collider.list->dim.worldSphere;
+    worldSphere->center.x = this->actor.posRot.pos.x;
+    worldSphere->center.y = (s32)(this->actor.posRot.pos.y + 59.5f);
+    worldSphere->center.z = this->actor.posRot.pos.z;
+}
+
+void func_808979C0(BgJyaGoroiwa* this, GlobalContext* ctx) {
+    ColliderJntSph* colliderJntSph = &this->collider;
+
+    Collider_InitJntSph(ctx, colliderJntSph);
+    Collider_SetJntSph(ctx, colliderJntSph, &this->actor, &sJntSphInit, this->colliderItems);
+    func_80897970(this);
+    this->collider.list->dim.worldSphere.radius = 0x3A;
+}
+
+void func_80897A2C(BgJyaGoroiwa* this) {
+    Actor* thisx = &this->actor;
+
+    f32 rotFactor = 175.30046f;
+    f32 posDiff = thisx->posRot.pos.x - thisx->pos4.x;
```

Comment:
Normally we'd put all the declarations together, followed by an empty line. I get why you spaced this the way you did, though. Idk.